mkosi/tree.py

```diff
@@ -61,4 +76,21 @@ def btrfs_maybe_snapshot_subvolume(config: MkosiConfig, src: Path, dst: Path) ->
         result = 1
 
     if result != 0:
-        copy_path(src, dst)
+        run(copy)
+
+
+def move_tree(config: MkosiConfig, src: Path, dst: Path) -> None:
+    if src == dst:
+        return
+
+    if dst.is_dir():
+        dst = dst / src.name
+
+    try:
+        src.rename(dst)
+        return
+    except OSError as e:
+        if e.errno != errno.EXDEV:
+            raise e
+
+    copy_tree(config, src, dst)
```

Comment:
```suggestion
        if e.errno != errno.EXDEV:
            raise e

        copy_tree(config, src, dst)
```
to make it visually a bit clearer, that this will only be hit in the exception case. That being said, the semantic is a bit weird, since in the except case the source will still be there, no?

Reply:
Oops I added an `rm -rf` to remove the tree when we copy it.